### DISCUSSION TITLE
Add facility to pass parameters to Forms

### DISCUSSION THREAD

**Discussion by YogiYang** [twinbasic/twinbasic#403] [Ideas]
We all as developers are familiar with procedures and functions which accept parameters.

In the say way if we can add support for defining parameters at Form level it would be a killer feature.

I don't know if I will be able to explain this feature properly or not but let me give it a try.

When we want to pass data between forms we currently declare **Global** variables in a module and then access those variables. 

And we all know that such methods can and will generate a lot of bugs especially when there is a team of developers working on the same project. Generally developers tend to forget that they have to first populate such and such a variable before calling/loading such and such a form and this leads to many unwanted bugs and hair splitting as well as time consuming debugging activities.

Again such **Global** variables used just for passing data between forms have no other use and they keep on popping up in IntelliSense when we are coding.

If there is a facility to define parameters in a form then we will not have to create a plethora of such **Global** variables. This again will result into peace of mind. And when a developer is loading/calling a form s/he will know in advance that the called form requires a set of data/parameters and the data if not provided it can be _detected at compile time_. 

Do think on this feature. It would be just great according to my perception & understanding.

**Comment by Kr00l** [comment]
I'm sure Wayne will add something similar to forms like already done for classes where there is the legacy Class_Initialize and the new parameterized Sub New.
